modules/extension/ysld/src/test/java/org/geotools/ysld/parse/YsldParseCookbookTest.java

```diff
@@ -604,6 +606,45 @@ public void testPointWithRotatedLabel() throws Exception {
         assertEquals(-45, Filters.asInt(placement.getRotation()));
     }
 
+    @Test
+    public void testPointWithRuleVendorOption() throws Exception {
+        // <UserStyle>
+        //   <Title>Simple Point With Rule Vendor Option</Title>
+        //   <FeatureTypeStyle>
+        //     <Rule>
+        //       <PointSymbolizer>
+        //         <Graphic>
+        //           <Mark>
+        //             <WellKnownName>circle</WellKnownName>
+        //             <Fill>
+        //               <CssParameter name="fill">#FF0000</CssParameter>
+        //             </Fill>
+        //           </Mark>
+        //           <Size>6</Size>
+        //         </Graphic>
+        //       </PointSymbolizer>
+        //       <VendorOption name="inclusion">mapOnly</VendorOption>
+        //     </Rule>
+        //     <Rule>
+        //       <PointSymbolizer>
+        //         <Graphic>
+        //           <ExternalGraphic>
+        //              <OnlineResource xlink:type="simple" xlink:href="smileyface.png" />
+        //             <Format>image/png</Format>
+        //           </ExternalGraphic>
+        //           <Size>20</Size>
+        //         </Graphic>
+        //       </PointSymbolizer>
+        //       <VendorOption name="inclusion">legendOnly</VendorOption>
+        //     </Rule>
+        //   </FeatureTypeStyle>
+        // </UserStyle>
```

Comment:
The cookbook tests are here to ensure the examples from the documentation work as advertised, so one fix is to add this to the cookbook.